integration/src/wallets/vultisig.ts

```diff
@@ -0,0 +1,413 @@
+          return Promise.resolve({
+            encoded: Buffer.from(mockTx, "hex"),
+          });
+        }
+        case "get_address":
+          return Promise.resolve("1FH6ehAd5ZFXCM1cLGzHxK1s4dGdq1JusM");
+        case "get_public_key":
+          return Promise.resolve("02a0434d9e47f3c86235477c7b1ae6ae5d3442d49b1943c2b752a68e2a47e247c7");
+        default:
+          throw new Error(`utxo: Unknown method ${method}`);
+      }
+    }),
+  } as unknown as VultisigUtxoProvider);
+
+const solanaProvider = {
+  publicKey: undefined,
+  connect: jest.fn(() => Promise.resolve({ address: "mock-solana-address" })),
+  signTransaction: jest.fn(() => Promise.resolve("mock-transaction")),
+  signAndSendTransaction: jest.fn(() => Promise.resolve({ signature: "mock-signature" })),
+  getAccounts: jest.fn(() => Promise.resolve([{ address: "mock-solana-address" }])),
+} as unknown as VultisigSolanaProvider;
+
+const cosmosProvider = {
+  getOfflineSigner: jest.fn(() => ({
+    getAccounts: jest.fn(() =>
+      Promise.resolve([
+        {
+          address: "cosmos15cenya0tr7nm3tz2wn3h3zwkht2rxrq7q7h3dj",
+          pubkey: "A4ix9R88YyVHfBqurqIhcUjDBAm2gyC6y3K+SztgqFCJChQ=",
+        },
+      ])
+    ),
+    getPubkey: jest.fn(() =>
+      Promise.resolve(
+        new Uint8Array([
+          2, 136, 177, 245, 49, 184, 120, 113, 149, 190, 179, 162, 33, 113, 72, 195, 4, 9, 54, 131, 200, 139, 114, 190,
+          75, 59, 96, 168, 80, 137, 10, 33, 42,
+        ])
+      )
+    ),
+  })),
+} as unknown as VultisigBftProvider;
+
+const thorchainProvider = {
+  getOfflineSigner: jest.fn(() => ({
+    getAccounts: jest.fn(() =>
+      Promise.resolve([
+        {
+          address: "thor1ls33ayg26kmltw7jjy55p32ghjna09zp74t4az",
+          pubkey: "A4ix9R88YyVHfBqurqIhcUjDBAm2gyC6y3K+SztgqFCJChQ=",
+        },
+      ])
+    ),
+    getPubkey: jest.fn(() =>
+      Promise.resolve(
+        new Uint8Array([
+          2, 136, 177, 245, 49, 184, 120, 113, 149, 190, 179, 162, 33, 113, 72, 195, 4, 9, 54, 131, 200, 139, 114, 190,
+          75, 59, 96, 168, 80, 137, 10, 33, 42,
+        ])
+      )
+    ),
+  })),
+} as unknown as VultisigBftProvider;
+
+export function createInfo(): core.HDWalletInfo {
+  return new vultisig.VultisigHDWalletInfo(ethereumProvider);
+}
+
+export async function createWallet(): Promise<core.HDWallet> {
+  const wallet = new vultisig.VultisigHDWallet(
+    ethereumProvider,
+    createUtxoProvider("bitcoin"), // bitcoinProvider
+    createUtxoProvider("litecoin"), // litecoinProvider
+    createUtxoProvider("dogecoin"), // dogecoinProvider
+    createUtxoProvider("bitcoincash"), // bitcoincashProvider
+    createUtxoProvider("zcash"), // zcashProvider
+    createUtxoProvider("dash"), // dashProvider
+    solanaProvider,
+    thorchainProvider, // thorchainProvider
+    cosmosProvider // cosmosProvider
+  );
+
+  wallet.cosmosSignTx = jest.fn(async () => {
```

Comment:
q: do we want those overrides? Isn't that effectively voiding the guarantees of testing those methods?